### PR TITLE
gateway requests reaching ui handler to fail gracefully

### LINKED ISSUE #1575: Non existing routes to return json 404 response instead of xml
See failure to parse response in issue #1574

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1575

### CODE REVIEW

File: api/ui_handler.go [hunk 1]
```diff
@@ -22,6 +25,12 @@ func NewUIHandler(authService auth.Service) http.Handler {
 
 func NewHandlerWithDefault(root http.FileSystem, handler http.Handler, defaultPath string) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if s3auth := r.Header.Get(sig.V4authHeaderName); strings.HasPrefix(s3auth, sig.V4authHeaderPrefix) {
```
Comment (ozkatz) [inline api/ui_handler.go:28]: For sigv4 this'll work, however, I suggest also checking `s3auth` against `sig.V2AuthHeaderRegexp` to cover all bases.

File: gateway/errors/errors.go [hunk 2]
```diff
@@ -745,4 +746,9 @@ var Codes = errorCodeMap{
 		Description:    "This operation is not supported in LakeFS",
 		HTTPStatusCode: http.StatusMethodNotAllowed,
 	},
+	ERRLakeFSWrongEndpoint: {
+		Code:           "ERRLakeFSWrongEndpoint",
+		Description:    "S3 request received in UI handler, did you forget to set your s3 gateway domain name?",
```
Comment (ozkatz) [inline gateway/errors/errors.go:751]: We can get the currently configured domain name from config, assuming we have access to it, to make this error a little friendlier (i.e. `S3 request received by UI handler, Did you configure your S3 client to use the lakeFS endpoint ("${s3_endpoint_url")?`)

**Review by ozkatz — APPROVED** [review]
Looks good, I do suggest at least checking for sigv2

File: api/ui_handler.go (hunk 1 above)
Comment (itaiad200) [reply api/ui_handler.go]: Done.

File: gateway/errors/errors.go (hunk 2 above)
Comment (itaiad200) [reply gateway/errors/errors.go:751]: Done.

**Review by ozkatz — APPROVED** [review]

**Pull request merged**